Support Organizations that do not use AWS Control Tower

gcasilva opened the pull request:
Adding Support Organizations that do not use AWS Control Tower

gcasilva committed:
Fixes to allow Non Control Tower environment deployment

gcasilva committed:
Fix entrypoint

gcasilva committed:
Updating Cleanup script

gcasilva commented:
@jefferyfry Like we talked after reviewing this please add to your Lacework ABI documentation:
“If you're using this solution in an AWS organization that doesn't use AWS Control Tower, before deploying this ABI solution you need to make sure:
1 - you have enabled Amazon SNS notification in your CloudTrail trail if using this service integration and,
2 - you need to create IAM roles to [Set up basic permissions for stack set operations](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-prereqs-self-managed.html#stacksets-prereqs-accountsetup): 
    a. You need to create `pAdminRoleName` [Parameter used while launching the solution] in your management account. The CloudFormation template to create this role is [available here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-prereqs-self-managed.html#stacksets-prereqs-accountsetup).
    b. You need to create `pExecRoleName` [Parameter used while launching the solution] across all child accounts with in the organization. You can use [this CloudFormation template](https://s3.amazonaws.com/cloudformation-stackset-sample-templates-us-east-1/AWSCloudFormationStackSetExecutionRole.yml) and deploy the stack across the organization using instructions from [Create a stack set with service-managed permissions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-getting-started-create.html#stacksets-orgs-associate-stackset-with-org)”

kkvinjam commented on .taskcat.yml:
```diff
@@ -17,4 +17,30 @@ tests:
+      pControlTower: 'false'
+      pLogArchiveAccountId: $[taskcat_ssm_/nonct/log-archive-account-id]
+      pSecurityAccountId: $[taskcat_ssm_/nonct/audit-account-id]
+      pGovernedRegions: 'us-east-1'
```
Avoid hard-coding values where ever possible between lines 37-46.

gcasilva replied:
Resolved in last commit.

kkvinjam commented on templates/cfn-abi-lacework-polygraph.template.yaml:
```diff
@@ -749,7 +833,11 @@ Resources:
     Properties:
-      TemplateURL: !Sub "https://${pSRASourceS3BucketName}.s3.${pSRAS3BucketRegion}.${AWS::URLSuffix}/${pSRAStagingS3KeyPrefix}/templates/cfn-abi-lacework-sec-hub-ingest.template.yaml"
+      TemplateURL: 
+        !If
```
Why is this needed? both templates are same.

kkvinjam commented on templates/cfn-abi-lacework-polygraph.template.yaml:
```diff
@@ -698,6 +762,17 @@ Resources:
+        AdministrationRoleARN: 
+          !If
+            - cControlTowerEnabled
+            - !Join [':', ['arn:aws:iam:', !Ref 'AWS::AccountId', 'role/service-role/AWSControlTowerStackSetRole']]
```
for lines 768-769, do not hard-code partition, use AWS::Partition instead.

Also, consider simpler representation as stated below:
```
- !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/service-role/AWSControlTowerStackSetRole
- !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:role/${pAdminRoleName}

gcasilva replied:
Resolved in last commit.

gcasilva committed:
Improvements on Kishore's comments

gcasilva commented:
@jefferyfry Can you please review approve this PR to add support for non-control tower environments?

jefferyfry reviewed (approved):
LGTM

gcasilva commented:
/do-e2e-tests

gcasilva committed:
Update cfn-abi-lacework-polygraph.template.yaml

Fix linting error on cfn-abi-lacework-polygraph.template.yaml

gcasilva committed:
Update cfn-abi-lacework-polygraph.template.yaml

Update Lambda python runtime to 3.11

gcasilva committed:
Update entrypoint.sh

Removed scoutsuite for testing

gcasilva commented:
/do-e2e-tests

gcasilva commented:
/do-e2e-tests

gcasilva commented:
/do-e2e-tests

gcasilva committed:
Merge branch 'main' into main

gcasilva committed:
Update entrypoint.sh

gcasilva commented:
/do-e2e-tests

gcasilva commented:
/do-e2e-tests

gcasilva committed:
Update entrypoint.sh to retrigger test

gcasilva committed:
Update entrypoint.sh 

Non-Control Tower environment update + add CloudTrail SRA Organizations test

gcasilva committed:
Update .taskcat.yml - update to support CloudTrail SRA Org test

update to support CloudTrail SRA Org test

gcasilva committed:
Update cleanup_config.json

Update cleanup config to support Cloudtrail SRA creation

gcasilva committed:
Update cfn-abi-lacework-polygraph.template.yaml

Update to support CloudTrail SRA Organizations creation

gcasilva committed:
Update Lacework Control Tower submodule to latest HEAD

gcasilva committed:
Point Lacework Control Tower submodule to PR #9

gcasilva committed:
Update SRA submodule to latest HEAD

gcasilva commented:
/do-e2e-tests

gcasilva committed:
Update .taskcat.yml - non control tower test

Update taskcat - non control tower test to add guardduty and securityhub creation

gcasilva committed:
Update cleanup_config.json

Update cleanup_config.json - to clean sra org trail s3 bucket

gcasilva commented:
/do-e2e-tests

gcasilva commented:
/do-e2e-tests

gcasilva commented:
/do-e2e-tests
